In [2]:
import pandas as pd

# Load data
X_train_resampled = pd.read_csv("X_train_resampled.csv")
y_train_resampled = pd.read_csv("y_train_resampled.csv")

X_test = pd.read_csv("X_test.csv")
y_test = pd.read_csv("y_test.csv")

In [3]:
y_train_resampled = y_train_resampled.values.ravel()
y_test = y_test.values.ravel()

In [4]:
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import accuracy_score, classification_report, confusion_matrix, roc_auc_score

lr_model = LogisticRegression(max_iter=1000, random_state=42)
lr_model.fit(X_train_resampled, y_train_resampled)

y_pred_lr = lr_model.predict(X_test)
y_prob_lr = lr_model.predict_proba(X_test)[:, 1]

print("Accuracy:", accuracy_score(y_test, y_pred_lr))
print("Confusion Matrix:\n", confusion_matrix(y_test, y_pred_lr))
print("Classification Report:\n", classification_report(y_test, y_pred_lr))
print("ROC-AUC:", roc_auc_score(y_test, y_prob_lr))

Accuracy: 0.9733682103858713
Confusion Matrix:
 [[55355  1509]
 [    8    90]]
Classification Report:
               precision    recall  f1-score   support

           0       1.00      0.97      0.99     56864
           1       0.06      0.92      0.11        98

    accuracy                           0.97     56962
   macro avg       0.53      0.95      0.55     56962
weighted avg       1.00      0.97      0.98     56962

ROC-AUC: 0.970028112187475


📄 Logistic Regression Results Analysis

The Logistic Regression model achieved an accuracy of 97.3% and a ROC-AUC score of 0.97.

However, due to the imbalanced nature of the dataset, accuracy alone is not a reliable metric.

The model achieved a high recall (0.92) for fraudulent transactions, meaning it successfully detected most fraud cases.

However, the precision is very low (0.06), indicating that many normal transactions were incorrectly classified as fraud.

This results in a high number of false positives, which may not be practical in real-world applications.